src/renderer/patches/screenShareFixes.ts

```diff
@@ -11,21 +11,70 @@ import { isLinux } from "renderer/utils";
 
 const logger = new Logger("VesktopStreamFixes");
 
-if (isLinux) {
-    const original = navigator.mediaDevices.getDisplayMedia;
-
-    async function getVirtmic() {
-        try {
-            const devices = await navigator.mediaDevices.enumerateDevices();
-            const audioDevice = devices.find(({ label }) => label === "vencord-screen-share");
-            return audioDevice?.deviceId;
-        } catch (error) {
-            return null;
-        }
+const original = navigator.mediaDevices.getDisplayMedia;
+
+interface OverrideDevices {
+    audio: string | undefined;
+    video: string | undefined;
+}
+
+let overrideDevices: OverrideDevices = { audio: undefined, video: undefined };
+
+export const patchOverrideDevices = (newOverrideDevices: OverrideDevices) => {
+    overrideDevices = newOverrideDevices;
+};
+
+async function getVirtmic() {
+    try {
+        const devices = await navigator.mediaDevices.enumerateDevices();
+        const audioDevice = devices.find(({ label }) => label === "vencord-screen-share");
+        return audioDevice?.deviceId;
+    } catch (error) {
+        return null;
+    }
+}
+
+navigator.mediaDevices.getDisplayMedia = async function (opts) {
+    const stream = await original.call(this, opts);
+
+    if (overrideDevices.audio) {
```

Comment:
this entire block is strange. why get temporary video & audio tracks just to remove them again and call getUserMedia for a second time? you can just set deviceId and similar properties on the options object before calling the original